draft-ZIP-0227.rst

```diff
@@ -0,0 +1,184 @@
+
+We have designed the protocol such that at the time of issuance, the issuer can already allocate all the tokens to the corresponding owners by creating the corresponding (shielded) output notes to the corresponding addresses. As is implied, the issuance mechanism is itself transparent, but the allocation is totally private. Furthermore, in a single issuance transaction, the issuer can create multiple asset types (and allocate all of those too). Finally, a key component of our design is the `finiteIssuance` boolean that defines whether a specific asset type can have more tokens issued or not.
+
+The issuance mechanism is broad enough for issuers to either create native assets on Zcash (i.e.: assets that originate on the Zcash blockchain), as well as for institutions to create bridges from other chains and import wrapped assets. In fact, the protocol described below enables, what we hope will be a useful set of, applications:
+
+- By setting the `finiteIssuance = 1` from the first issuance instance of that asset type, the issuer is in essence creating a one-time issuance transaction. This is useful when the max supply is capped from the beginning and the distribution is known in advance. All tokens are issued at once and distributed as needed.
+- When the `finiteIssuance = 0`, the issuer can keep creating tokens of that type (in a transparent manner). The boolean value can be changed with any new issuance transaction, and could be set to `1` either when the issuer keys have been compromised, and hence stopping all issuances of that asset type (the boolean cannot be reversed), or whenever the issuer decides that the max supply has been reached.
+- Note that this mechanism can be used with the burning process to control and affect the supply of any custom asset.
+
+
+Asset Identifier Generation
+===========================
+
+For every new asset, there must be a new and unique identifier of the asset. Our design decision is not to have a chosen name for the asset type, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”. From this asset identifier, which is a global byte string (valid across all Zcash protocols), we derive the specific type used within the notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve)
+
+The main requirement for the asset identifier is for it to be collision-free, as two different issuers should not be able to issue the same asset type. As such we need the asset identifier to be:
+
+- pseudorandom
+- deterministic
+- dependent on the issuer ID
+
+We define the asset identifier `assetID` (represented as a string) as
+
+:math:`\mathsf{assetID := GroupHash^{\mathbb{P}}}\mathsf{(issuerID || assetDesc)}`
```

Comment:
What kind of string is the assetID? The output type of GroupHash is a Pallas curve group element. Hex encoding?

Reply:
I can resolve the first part of the comment:
After the code was written (and with the consequent code review) we changed the notation and specified much more clearly the difference between `asset_desc` which is a string and `AssetId` which is the actual group element used in the base point of the value commitment; and their relationship is specifed.

For the specific abstraction of the GroupHash into AssetIdHash, I believe it should be discussed further. At the moment this abstraction may not be needed.

Reply:
@paullaux this has not been resolved in that more discussion is needed to justify using a new definition of the asset ID hasher.

Reply:
The structure has been changed such that the asset identifier and asset description string are global (aka unchanging across potential future protocols), and the asset base is the term used for the element of the group curve, which would change between protocols